eng/scripts/Invoke-DependencyCheck.ps1

```diff
@@ -0,0 +1,77 @@
+Param(
+    [string] $newPackageDirectory
+)
+
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+$ignoreCheck = "sdk/synapse/azartifacts", "sdk/template/aztemplate", "sdk/azcore", "sdk/azidentity"
+
+$sdks = Get-AllPackageInfoFromRepo
+
+$sdkRoot = Join-Path $RepoRoot "sdk"
+
+# Get all existed packages
+$packagesImport = ""
+$newPackage = ""
+foreach ($sdk in $sdks)
+{
+    if ($sdk.Name -like "*internal*" -or $sdk.Name -in $ignoreCheck)
+    {
+        continue
+    }
+    if ((Resolve-Path $sdk.DirectoryPath).Path -eq (Resolve-Path (Join-Path $sdkRoot $newPackageDirectory)).Path)
```

Comment:
Do we really need to resolve the paths? Why not just compare the Name?

Reply:
As path separator may be different for `$sdk.DirectoryPath` and `$newPackageDirectory`. Resolve path is just for safety.

Reply:
I think comparing sdk.Name to newPackageDirectory should always work but if not then perhaps doing a simple replace of "\\" with "/" at the top of this file should work. It will be a lot more efficient then resolving the paths every time.

Reply:
Fixed.